products/BellHybrid/apps/application-bell-settings/models/alarm_settings/AlarmSettingsListItemProvider.cpp

```diff
@@ -69,8 +73,44 @@ namespace app::bell_settings
 
         internalData.emplace_back(alarmVolume);
 
-        internalData.emplace_back(
-            new OnOffListItem(model.getAlarmLightOnOff(), utils::translate("app_bell_settings_alarm_settings_light")));
+        auto onOffLight = new OnOffListItem(settingsModel.getAlarmLightOnOff(),
+                                            utils::translate("app_bell_settings_alarm_settings_light"));
+
+        internalData.emplace_back(onOffLight);
```

Comment:
This is not exception-safe but since when an exception is thrown the device is restarted anyway I guess it may be left as it is.

Another thing is that I don't like the fact that `internalData` is modified here directly when this is a container of pointers managed by one of the base classes. It's because someone may do anything with this container, e.g. remove some pointers and the objects will not be destroyed when base class is destroyed. But changing it is probably outside of the scope of this PR so I'm ok with leaving it as it is.

Reply:
I totally agree with you. However, I tried to follow the old base of code and don't change a lot. 
As you said before - it is not part of this PR, but definitely, it should be addressed and changed in the feature.
On the other hand, almost every application on our system has an MVC pattern so we need to redesign every `provider`.